server/src/main/java/org/opensearch/index/store/remote/utils/TransferManager.java

```diff
@@ -82,14 +85,18 @@ public IndexInput fetchBlob(BlobFetchRequest blobFetchRequest) throws IOExceptio
             });
         });
 
-        // Cache entry was either retrieved from the cache or newly added, either
-        // way the reference count has been incremented by one. We can only
-        // decrement this reference _after_ creating the clone to be returned.
-        try {
-            return cacheEntry.getIndexInput().clone();
-        } finally {
-            fileCache.decRef(key);
-        }
+        return AccessController.doPrivileged((PrivilegedAction<IndexInput>) () -> {
```

Comment:
@finnegancarroll could we combine this with the previous `AccessController.doPrivileged` block?

```
return AccessController.doPrivileged((PrivilegedAction<IndexInput>) () -> {
            var cacheEntry = fileCache.compute(key, (path, cachedIndexInput) -> {
                if (cachedIndexInput == null || cachedIndexInput.isClosed()) {
                    logger.trace("Transfer Manager - IndexInput closed or not in cache");
                    // Doesn't exist or is closed, either way create a new one
                    return new DelayedCreationCachedIndexInput(fileCache, streamReader, blobFetchRequest);
                } else {
                    logger.trace("Transfer Manager - Already in cache");
                    // already in the cache and ready to be used (open)
                    return cachedIndexInput;
                });
            try {
                return cacheEntry.getIndexInput().clone();
            } catch (IOException e) {
                throw new RuntimeException(e);
            } finally {
                fileCache.decRef(key);
            }
        });
```